roles/contiv_network/tasks/main.yml

```diff
@@ -1,44 +1,10 @@
 ---
 # This role contains tasks for configuring and starting netmaster and netplugin service
-
-- name: download netmaster and netplugin
-  get_url:
-    validate_certs: no
-    url: https://github.com/contiv/netplugin/releases/download/v0.0.0-12-11-2015.20-54-40.UTC/netplugin-v0.0.0-12-11-2015.20-54-40.UTC.tar.bz2
-    dest: /tmp/contivnet.tar.bz2
-
-- name: install netmaster and netplugin
-  shell: tar vxjf /tmp/contivnet.tar.bz2
-  args:
-    chdir: /usr/bin/
-    creates: netmaster
-
-- name: copy environment file for netplugin
-  template: src=netplugin.j2 dest=/etc/default/netplugin
-
-- name: copy systemd units for netplugin
-  copy: src=netplugin.service dest=/etc/systemd/system/netplugin.service
-
-- name: start netplugin
-  shell: systemctl daemon-reload && systemctl start netplugin
-
-- name: setup netmaster host alias
-  shell: echo "{{ service_vip }} netmaster" >> /etc/hosts
-
-- name: copy environment file for netmaster
-  copy: src=netmaster dest=/etc/default/netmaster
-  when: run_as == "master"
-
-- name: copy systemd units for netmaster
-  copy: src=netmaster.service dest=/etc/systemd/system/netmaster.service
-  when: run_as == "master"
-
-# resolves an issue with netmaster: https://github.com/contiv/netplugin/issues/210
-- name: start netmaster
-  shell: systemctl daemon-reload && systemctl start netmaster
-  when: run_as == "master"
-  run_once: true
-
-# FIXME: need to move the following to correct roles -- also needs release tooling
-- name: install contivctl
-  shell: source /etc/profile.d/00golang.sh && go get github.com/contiv/contivctl
+- name: pull dns container image
+  docker:
```

Comment:
So, this is the logs from the second run of provisioning for the nodes. Shell module seems to be repulling the image even if it exists. Also, this might be important, because a lot of the complaints that I have been hearing is because of re-pulling this image during vagrant bringup:
When using docker module
TASK: [contiv_network | pull dns container image] *****************************
ok: [netplugin-node1]
ok: [netplugin-node2]

When using shell module:
TASK: [contiv_network | pull dns container image] *****************************
changed: [netplugin-node2]
changed: [netplugin-node1]

The second reason is that like I mentioned, I will definitely have to reintroduce this docker module within a week when aci ansible changes come in. We will definitely remove this when we move it to a service model. Till then I prefer keeping this.